iscript/src/iscript/mac.py

```diff
@@ -143,8 +143,26 @@ def get_bundle_executable(appdir):
 
 
 # _get_sign_command {{{1
-def _get_sign_command(identity, keychain, sign_config):
-    return ["codesign", "-s", identity, "-fv", "--keychain", keychain, "--requirement", sign_config["designated_requirements"] % {"subject_ou": identity}]
+def _get_sign_command(identity, keychain, sign_config, file_=None, entitlements_path=None):
+    sign_command = [
+        "codesign",
+        "-s",
+        identity,
+        "-fv",
+        "--keychain",
+        keychain,
+        "--requirement",
+        sign_config["designated_requirements"] % {"subject_ou": identity},
+    ]
+
+    if file_ and file_ in sign_config.get("hardened_runtime_files", []):
+        sign_command.extend(["-o", "runtime"])
+    elif sign_config.get("sign_with_entitlements", False) and (file_ and file_ not in sign_config.get("no_entitlements_files", [])):
```

Comment:
Is there a reason we have both `hardened_runtime_files` and `no_entitlements_files`? (Will we need some files with no `-o runtime` at some point?)

Similarly, since these blocks are mutually exclusive, is there a reason why `wg` and `wireguard-go` are in both `hardened_runtime_files` and `no_entitlements_files` ?

If we see needing more complex special configs per file, we may want a dictionary keyed by filename. We don't have to go down that route unless we see it coming shortly. I mainly want to make sure we want two config settings here, when it seems like one would solve the issue at hand.

Reply:
I can try to simplify this, I agree it's not great, and it ended up this why through evolution, not design...

Truthfully though, I'm treating this is a stop-gap solution until we have a proper manifest for specifying these things for an entire app (ideally one that comes in as part of of the task payload).

Reply:
LMK. If it turns out this is the best option for reasons, we can land as is. If there's a quick way to improve it, let's do that.

Reply:
Okay, so after some review I believe the following is true:
* We have to keep `no_entitlements_files` (or something equivalent), because we cannot sign two of the vpn binaries with entitlements.
* If we want to preserve the behaviour of _not_ signing with a hardened runtime by default, we have to keep `hardened_runtime_files` (to enable it for these same files without enabling entitlements). The alternative is to always include `-o runtime` in the signing command, which may or may not break something.

Do you have a preference here?

Reply:
> Okay, so after some review I believe the following is true:
> 
>     * We have to keep `no_entitlements_files` (or something equivalent), because we cannot sign two of the vpn binaries with entitlements.

The `no_entitlements_files` check is in an `elif` where we've already hit matched the `if` block (`hardened_runtime_files`). As far as I can see, nothing uses `no_entitlements_files` without `hardened_runtime_files`. Do you think we'll still need both in some future use case?

Reply:
(Marked approved. If the answer is `no`, we can remove `no_entitlements_files`. If the answer is `yes`, we can land as is.)

Reply:
Oh, I see. Yes, we could get rid of `no_entitlements_files`, but I'd probably rename `hardened_runtime_files` to `hardened_runtime_only_files` for clarity (otherwise one might expect them to still get entitlements, and anything else _not_ to have hardened runtime).